Public/Src/IDE/Debugger/DebuggerState.cs

```diff
@@ -66,7 +82,7 @@ public DebuggerState(PathTable pathTable, LoggingContext loggingContext, Logger
         /// <returns>List of removed ThreadId -> ThreadState key value pairs.</returns>
         public IReadOnlyDictionary<int, ThreadState> ClearStoppedThreads()
         {
-            lock (this)
+            lock (m_lock)

```

Comment:
>lock (m_lock) [](start = 12, length = 13)

What is the reason to lock this "object" m_lock instead of locking "this" as was done before?

Reply:
Best practices recommend to use a private object for such locking.  The idea is that if you lock on `this`, and the `this` object is accessible to others (which it always is), then someone else can also use it for locking and potentially cause deadlocks.